PurchasesCoreSwift/Identity/IdentityManager.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  IdentityManager.swift
+//
+//  Created by Joshua Liebowitz on 8/9/21.
+
+import Foundation
+
+// TODO (post-migration): Change back to internal, including all public properties.
+@objc(RCIdentityManager) public class IdentityManager: NSObject {
+
+    @objc public var currentAppUserID: String? {
+        return deviceCache.cachedAppUserID
+    }
+
+    @objc public var currentUserIsAnonymous: Bool {
+        let anonymousRegex = #"\$RCAnonymousID:([a-z0-9]{32})$"#
+        let anonymousFoundRange = currentAppUserID?.range(of: anonymousRegex, options: .regularExpression)
+        let foundAnonymous = anonymousFoundRange != nil && !(anonymousFoundRange?.isEmpty ?? false)
```

Comment:
do we need the isEmpty? won't `anonymousFoundRange` be nil in EITHER the case that currentAppUserId is nil or the regex isn't found?

also maybe we can have a guard on currentAppUserId to early return

Reply:
You're right in all these cases, I'm cleaning this up now.